plugins/mumble_plugin_linux.h

```diff
@@ -0,0 +1,215 @@
+// Copyright 2005-2016 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_PLUGIN_LINUX_H_
+#define MUMBLE_PLUGIN_LINUX_H_
+
+#ifndef PTR_TYPE_CONCRETE
+# define PTR_TYPE_CONCRETE PTR_TYPE
+#endif
+
+#include <sys/uio.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string>
+#include <cstring>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+#include "mumble_plugin.h"
+
+pid_t pPid;
+static PTR_TYPE_CONCRETE pModule;
+
+static inline std::string readAll(std::string fn) {
+	std::ifstream ifs;
+	ifs.open(fn, std::ifstream::binary);
+
+	std::string content;
+
+	char buf[256];
+	while (ifs.good()) {
+		ifs.read(&buf[0], sizeof(buf));
+		size_t nread = ifs.gcount();
+		if (nread > 0) {
+			content.append(&buf[0], nread);
+		}
+	}
+
+	return content;
+}
+
+static inline PTR_TYPE_CONCRETE getModuleAddr(pid_t pid, std::string mod) {
+	std::string mapsFn = std::string("/proc/") + std::to_string(static_cast<unsigned long long>(pid)) + std::string("/maps");
+	std::string maps = readAll(mapsFn);
+
+	if (maps.size() == 0) {
+		return 0;
+	}
+
+	std::stringstream ss(maps);
+	std::string line;
+	while (ss.good()) {
+		std::string baseaddr;
+
+		int ch;
+		while (1) {
+			ch = ss.get();
+			if (ch == '-') {
+				break;
+			} else if (ch == EOF) {
+				return 0;
+			}
+			baseaddr.push_back(static_cast<char>(ch));
+		}
+
+		// seek to perms
+		do {
+			ch = ss.get();
+			if (ch == EOF) {
+				return 0;
+			}
+		} while (ch != ' ');
+
+		// seek to offset
+		do {
+			ch = ss.get();
+			if (ch == EOF) {
+				return 0;
+			}
+		} while (ch != ' ');
+
+		// seek to dev
+		do {
+			ch = ss.get();
+			if (ch == EOF) {
+				return 0;
+			}
+		} while (ch != ' ');
+
+		// seek to inode
+		do {
+			ch = ss.get();
+			if (ch == EOF) {
+				return 0;
+			}
+		} while (ch != ' ');
+
+		// seek to pathname
+		do {
+			ch = ss.get();
+			if (ch == EOF) {
+				return 0;
+			}
+		} while (ch != ' ');
+
+		// eat spaces until we're at the beginning of pathname.
+		while (ch == ' ') {
+			if (ch == EOF) {
+				return 0;
+			}
+			ch = ss.get();
+		}
+		ss.unget();
+
+		std::string pathname;
+		while (1) {
+			ch = ss.get();
+			if (ch == '\n') {
+				break;
+			} else if (ch == EOF) {
+				return 0;
+			}
+			pathname.push_back(static_cast<char>(ch));
+		};
+
+		// OK, we found 'em!
+		// Only treat path as a real path if it starts with /.
+		if (pathname.size() > 0 && pathname.at(0) == '/') {
+			// Find the basename.
+			size_t lastSlash = pathname.find_last_of('/');
+			if (pathname.size() > lastSlash + 1) {
+				std::string basename = pathname.substr(lastSlash + 1);
+				if (basename == mod) {
+					unsigned long long addr = std::stoull(baseaddr, 0, 16);
```

Comment:
Please use

    unsigned long addr = strtoul(baseaddr.c_str(), NULL, 16);

I am not sure if this will need a cast due to the return type of this function.